### DISCUSSION TITLE
preserve mobile application storage directory?

### DISCUSSION THREAD

**Discussion by AbigailGosling** [BowlerHatLLC/vscode-as3mxml#588] [Q&A]
Trying to migrate over from Flash Builder and I have the IDE mostly set up fine now. 

Final hurdle for android development in `File.applicationStorageDirectory` as it currently stand when I install and attach the existing debug build is removed from the android device before the new apk is installed. This wipes out all data stored in `File.applicationStorageDirectory`. Is there anyway to launch the debugger to do an update and debug rather than a reinstall and debug? Flash Builder could do this and it made fixing issues related to persistent data much easier.

My current launch.json

```
{
	// Use IntelliSense to learn about possible SWF debug attributes.
	// Hover to view descriptions of existing attributes.
	// For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
	"version": "0.2.0",
	"configurations": [
		{
			"type": "swf",
			"request": "launch",
			"name": "Launch SWF"
		},
        {
            "type": "swf",
            "request": "attach",
            "name": "Install and Attach (Android)",
            "platform": "android",
            "connect": true
        }
	]
}```

**Comment by joshtynjala** [comment]
There actually isn't a way to install a new debug build of an AIR app on an iOS device while persisting its application storage. You must fully uninstall before installing a new debug build. However, as you're aware, it is possible on Android. I believe that it was easier for me to simply make both platforms consistent by having the same limitation, so it's not possible at this time.
